sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEvent.kt

```diff
@@ -0,0 +1,142 @@
+    private val eventSpans: MutableMap<String, ISpan> = HashMap()
+    private val breadcrumb: Breadcrumb
+    internal val callRootSpan: ISpan?
+    private var response: Response? = null
+
+    init {
+        val urlDetails = UrlUtils.parse(request.url.toString())
+        val url = urlDetails.urlOrFallback
+        val host: String = request.url.host
+        val encodedPath: String = request.url.encodedPath
+        val method: String = request.method
+
+        // We start the call span that will contain all the others
+        callRootSpan = hub.span?.startChild("http.client", "$method $url")
+
+        urlDetails.applyToSpan(callRootSpan)
+
+        // We setup a breadcrumb with all meaningful data
+        breadcrumb = Breadcrumb.http(url, method)
+        breadcrumb.setData("url", url)
+        breadcrumb.setData("host", host)
+        breadcrumb.setData("path", encodedPath)
+        breadcrumb.setData("method", method)
+        breadcrumb.setData("success", true)
+
+        // We add the same data to the root call span
+        callRootSpan?.setData("url", url)
+        callRootSpan?.setData("host", host)
+        callRootSpan?.setData("path", encodedPath)
+        callRootSpan?.setData("http.method", method)
+        callRootSpan?.setData("success", true)
+    }
+
+    /**
+     * Sets the [Response] that will be sent in the breadcrumb [Hint].
+     * Also, it sets the protocol and status code in the breadcrumb and the call root span.
+     */
+    fun setResponse(response: Response) {
+        this.response = response
+        breadcrumb.setData("protocol", response.protocol.name)
+        breadcrumb.setData("status_code", response.code)
+        callRootSpan?.setData("protocol", response.protocol.name)
+        callRootSpan?.setData("status_code", response.code)
+        callRootSpan?.status = SpanStatus.fromHttpStatusCode(response.code)
+    }
+
+    fun setProtocol(protocolName: String?) {
+        if (protocolName != null) {
+            breadcrumb.setData("protocol", protocolName)
+            callRootSpan?.setData("protocol", protocolName)
+        }
+    }
+
+    fun setRequestBodySize(byteCount: Long) {
+        if (byteCount > -1) {
+            breadcrumb.setData("request_content_length", byteCount)
+            callRootSpan?.setData("http.request_content_length", byteCount)
+        }
+    }
+
+    fun setResponseBodySize(byteCount: Long) {
+        if (byteCount > -1) {
+            breadcrumb.setData("response_content_length", byteCount)
+            callRootSpan?.setData("http.response_content_length", byteCount)
+        }
+    }
+
+    /**
+     * Sets the success flag in the breadcrumb and the call root span to false.
+     * Also sets the [errorMessage] if not null.
+     */
+    fun setError(errorMessage: String?) {
+        breadcrumb.setData("success", false)
+        callRootSpan?.setData("success", false)
+        if (errorMessage != null) {
+            breadcrumb.setData("error_message", errorMessage)
```

Comment:
This is probably the `reason` field? https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/

Reply:
I'm not sure. `reason` is a `Text that describes the status code`. I think it refers to things like `Not found` or `Bad Method`, rather than the error that caused that status code.
This `error_message` is the actual message returned by the Exception

Reply:
True, makes sense, lets keep it.